pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
> @therc Hm...Let me see, I'm not familiar with that part, but it looks like we could uniform them with only one type. I'm not sure who we should ask about this, let's try, :)

`AuthConfiguration` predates `AuthConfig` by about a year.  I am inclined to maintain our own serialization format to ensure stability, but a reflective unit test to keep the two types in sync would be valuable.  Looks like they've drifted.

@lavalamp Its definitely API-like.  As I recall, the serialization type was made to be compatible with an existing docker config file format.